karate-core/src/main/java/com/intuit/karate/core/Step.java

```diff
@@ -43,12 +43,22 @@
     private Table table;
     
     public String getDebugInfo() {
-        String scenarioName = StringUtils.trimToNull(scenario.getName());
-        String message = "feature: " + scenario.getFeature().getRelativePath();
-        if (scenarioName != null) {
-            message = message + ", scenario: " + scenarioName;
-        }
-        return message + ", line: " + line;        
+        String message = "feature: ";
+
+        if(isBackground()) {
+            String featureName = StringUtils.trimToNull(feature.getName());
+            message += feature.getRelativePath();
+            if (featureName != null) {
+                message = message + ", feature: " + featureName;
+            }
+        } else {
+                String scenarioName = StringUtils.trimToNull(scenario.getName());
+                message = "feature: " + scenario.getFeature().getRelativePath();
+                if (scenarioName != null) {
+                    message += ", scenario: " + scenarioName;
+                }
+            }
+        return message + ", line: " + line;     
     }
```

Comment:
I would like to simplify this as follows:

```java
    public String getDebugInfo() {
        String message = "feature: " + feature.getRelativePath();
        if (scenario != null) { // isBackground()
            message = message + ", scenario: " + StringUtils.trimToNull(scenario.getName());
        }
        return message + ", line: " + line;
    }
```

I can make the change, but in case you want the contributor credit, please update :)

Reply:
Ok, feel free to do the change!
I'm only more in favor of using isBackground() because that function eliminates the need for the comment that it's not background ;)

Reply:
ok I used `isBackground()` and made the change, thanks for the help !